src/ZPublisher/HTTPRequest.py

```diff
@@ -1204,11 +1204,11 @@ def resolve_url(self, url):
         # method is called on the response).
         try:
             object = req.traverse(path)
-        except Exception:
+        except Exception as exc:
             rsp.exception()
-        if object is None:
-            req.clear()
-            raise sys.exc_info()[0](rsp.errmsg)
+            if object is None:
+                req.clear()
+                raise exc.__class__(rsp.errmsg)
```

Comment:
This bit shouldn't be under the `except Exception as exc` block, as `object` won't be bound if `req.traverse(path)` raises.